File: c/hamming/src/hamming.c
#include "hamming.h"

long long compute(const char *const dna1, const char *const dna2)
{
    if (dna1 == NULL || dna2 == NULL)
    {
        return -1;
    }

    size_t difference = 0;
    size_t i = 0;
    while (dna1[i] != '\0' && dna2[i] != '\0')
    {
        if (dna1[i] != dna2[i])
        {
            difference++;
        }

        i++;
    }

    return dna1[i] == dna2[i] ? difference : -1;
}

File: javascript/hamming/hamming.js
class Hamming {
    constructor() {}

    compute(dna1, dna2) {
        if (dna1.length !== dna2.length) {
            throw new Error("DNA strands must be of equal length.");
        }

        return dna1.split('').filter((e, i) => e !== dna2[i]).length;
    }
};

module.exports = Hamming;
